backend/src/controllers/EventController.ts

```diff
@@ -268,6 +272,38 @@ export class EventController {
 
     return this.rsvpMapper.entityToResponse(newRSVP);
   }
+
+  @Delete('/:eventID/unrsvp/affiliate')
+  @ResponseSchema(RSVPResponse)
+  @OpenAPI({ security: [{ TokenAuth: [] }] })
+  async affiliateEventUnRSVP(
+    @Param('eventID') eventID: number,
+    @CurrentUser({ required: true }) appUser: AppUser
+  ): Promise<RSVPResponse | undefined> {
+    const getRSVPQuery = { eventId: eventID, appUserId: appUser.id };
+    const deletedRSVP = await this.rsvpService.deleteAffiliateRSVP(getRSVPQuery);
+
+    if (deletedRSVP === undefined) {
+      return undefined;
```

Comment:
should we return an error instead?

Reply:
I see. I'll take a look and make changes after finals week.